---
layout: single
title: "에러 발생 해결 방법"
categories:
  - Anaconda
classes: wide
tags:
  - jeykll
  - github
  - Minimalmistakes
  - blog
  - Anaconda
---

# 제가 실행하면서 발생했던 에러들을 해결하는 방법들을 계속 업데이트할 포스팅입니다.

### 1. 파이토치 가상환경 구성 pip install 중 에러 발생  
 + pip install certifi==2019.11.28 을 할 때 다음과 같은 에러 발생
 `ERROR: Cannot uninstall 'certifi'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.`

 해결 방법
 `pip install --ignore-installed certifi==2019.11.28`

 --ignore 를 추가하여 해결


### 2. 아톰에서 가상환경 접속하여 사용할 때 에러 발생
 + 가상환경 접속 후 Shift + Enter를 반복 사용하다보면 에러 발생
 `[IPKernelApp] ERROR | Exception in message handler: Traceback (most recent call last): File "/venv/.virtualenvs/flask_dash/lib/python3..... 등 엄청 장문의 에러 발생`


 해결 방법
 `pip install jedi==0.17.2`

 jedi 버전을 다운그레이드 후 해결

### 3. 학습을 시작할 때 발생하는 에러  
+ train.py 돌릴 때 발생하는 에러
`RuntimeError: Expected object of scalar type Long but got scalar type Int for argument #2 'target' in call to _thnn_nll_loss_forward`  

해결 방법  
`labels = labels.to(device)` 를 `labels = labels.to(device=device, dtype=torch.int64)` 로 변경  

+ `AttributeError: '_IncompatibleKeys' object has no attribute 'eval'`  

해결 방법  
`model = model.load_state_dict(checkpoint).to(DEVICE).eval()` 을  
`model = model.to(DEVICE)`  
`model.load_state_dict(checkpoint)`
`model.eval()` 로 변경  

+ `TypeError: softmax() missing 1 required positional argument: 'input'`

해결 방법  

softmax 함수를 torch.nn.functional.softmax() 썼는데 torch.nn.Softmax() 로 변경

### 4. matplotlib 에러  
+ git blog 에 있는 code demo 를 돌렸을 때 발생하였음
`'It is not currently possible to manually set the aspect '
NotImplementedError: It is not currently possible to manually set the aspect on 3D axes`  

해결 방법
`matplotlib 3 버전을 2.2버전으로 다운 그레이드`  
`pip install matplotlib==2.2.2`

### 5. scipy 에러  
+  git blog 에 있는 code demo 를 돌렸을 때 발생하였음  
`AttributeError: module 'scipy.misc' has no attribute 'imresize'`    

해결 방법  
`pip uninstall scipy`  
`pip install scipy==1.2.2` 하여 버전 다운 그레이드  
